File: sprint3/By lessons/lesson9code1/lesson9code1.go
package main

import (
    "fmt"
    "sort"
)

var digitLengths = []int{4, 4, 3, 3, 6, 4, 5, 4, 6, 6} // длины слов «ноль», «один»,...

func keyForCard(card int) []int {
    return []int{-digitLengths[card], card}
}

func main() {
    cards := []int{2, 3, 7}
    sort.Slice(cards, func(i, j int) bool {
        keyI := keyForCard(cards[i])
        keyJ := keyForCard(cards[j])
        if keyI[0] == keyJ[0] {
            return keyI[1] < keyJ[1]
        } 
        return keyI[0] < keyJ[0]
    })
    fmt.Println(cards)
}

File: sprint3/By lessons/lesson9code1/lesson9code1.swift
let digitLengths = [4, 4, 3, 3, 6, 4, 5, 4, 6, 6] // длины слов «ноль», «один»,...

func keyForCard(_ card: Int) -> [Int] {
    return [-digitLengths[card], card]
}

var cards = [2, 3, 7]
cards.sort(by: { card1, card2 in
    let key1 = keyForCard(card1)
    let key2 = keyForCard(card2)
    return key1.lexicographicallyPrecedes(key2)
})
print(cards)